### ISSUE TITLE
Improve documentation & searchability for v-on's .native modifier

### ISSUE THREAD

**Issue by alexsasharegan** [vuejs/vuejs.org#979]
### What problem does this feature solve?
# TL;DR;
- Makes understanding native event binding on custom Vue components more accessible. 
- Also reduces downstream issue load on component libraries.

As component libraries of custom components become more prevalent, authors have one of two choice: re-emit all/common events on the component, or educate users to use native binding modifier. I personally think that the `.native` modifier should become the component lib standard usage pattern because it reduces code size and the mental overhead of manually re-emitting all necessary events on a custom component.

As a contributor to the [bootstrap-vue](https://github.com/bootstrap-vue/bootstrap-vue/) component library, I know we get a decent amount of issues around this. We educate devs about the amazingly useful `.native` modifier, but ultimately are burdened by the limited searchability of current documentation.

### What does the proposed API look like?
The documentation on this is already sufficient, but it is not prominent. I know about this modifier, yet I still have a hard time finding it in the documentation (or google). I think the common use-case would lead a dev to search something like "listen to event on component" or "v-on not working on vue component". Having some valuable results turn up would be a big win for DX.

**Comment by chrisvfritz** [comment]
Thanks for bringing this up! I'll start by seeing what currently comes up for common searches and trying to add more hints and links to where the feature is documented.

On a related note, you may be interested in [this discussion](https://github.com/vuejs/vue/issues/5983) on the core Vue repo.

**Comment by alexsasharegan** [comment]
That discussion is very important! Thank you so much for the heads up. I'll clue in the bootstrap-vue team so they can weigh in any important thoughts from our experience.

**Comment by chrisvfritz** [comment]
The API was just improved with https://github.com/vuejs/vue/commit/61187596b9af48f1cb7b1848ad3eccc02ac2509d and these features will be documented as part of the release, so I think this is resolved now. 🙂
